include/xcshell/command_executor.h

```diff
@@ -14,13 +15,12 @@ class CommandExecutor {
  private:
   static std::vector<char *> BuildArgv(const std::string &command,
                                        const std::vector<std::string> &args);
-  static int ProcessChild(const std::string &command,
-                          const std::vector<std::string> &args);
+  static int ProcessChild(RedirectElement &redirectElement);
   static void WaitChildExit(pid_t pid);
-  // todo: remove basic type below using typedef or something else
-  std::tuple<std::string, std::vector<std::string>> ParseUserInput(
-      const std::string &input_line);
+
   BuildIn build_in_;
+
+  Parser parser;
```

Comment:
object member should be end with underscore `_`.

Reply:
This repo use Google C++ Code Style, maybe you can check it out.

Reply:
Do you init parser somewhere? I can not found. Looks you do not pass `build_in_` here to parser.